RELEASES.md

```diff
@@ -0,0 +1,78 @@
+# Releases Guide
+
+Here, we will briefly understand the release plan and versioning steps. 
+This will cover the release process, major and minor releases, backporting the PR and manual release steps.
+
+If there is something that you require or this document leaves out, please
+reach out by [filing an issue](https://github.com/coredns/coredns/issues).
+
+## Releases
+
+Releases will be created from the release branches. The release will follow the following convention as `<major>.<minor>.<patch>`. 
+All the major and minor releases will be made from the master. The tag will be of the format `v<major>.<minor>.<patch>`.
+
+After a minor release, a branch will be created with the format of `release-<major>.<minor>` from the minor tag. Once the branch is ready, all the patches will be released from that branch. 
+For example, once we release `v1.10.0`, a branch will be created as `release-1.10` with the following `v1.10` tag. Next, all future patches will be done against that branch. i.e. `v1.10.0, v1.10.1`.
+
+## Next Release
+
+The next release will be tracked in the milestone under the issue section, with Release tracking. 
+Also, it can be filtered with the label release. If your issue or PR is not present in a milestone, please create a PR to track the next release.
+
+## Coredns Support
+
+The support will be identified with several states:
+
+- __*Active*__: The release is a stable branch that is currently supported and accepting patches.
+- __*Maintenance*__: The release that is currently supported and accepting patches.
```

Comment:
suggest: This still does not clarify the differences between Active vs. Maintenance branches.

Reply:
As, I was not sure previously till when the support will be provided when the branch is not active. So, I kept in maintenance mode which is still accepting fixes. But now N, N-1 will be active releases as per the maintainers suggestion.